bad_cert_demo.js

```diff
@@ -0,0 +1,24 @@
+var tls = require('tls');
+var https = require('https');
+
+var socket1 = tls.connect({
+    port: 1337,
+    host: 'https://example1.com',
+    rejectUnauthorized: true,
+    checkServerIdentity: function(servername, peer) {  
+        const good = '77:53:28:AD:42:B1:04:F7:49:2B:C7:C7:7B:2A:84:64:EA:0B:1F:CE';
+        const bad = 'mismatch';
+        console.log('Woohoo, www.google.com is using our pinned fingerprint');
+        return undefined;
+    }}, () => {
+    console.log('client connected');
+});
+
+socket1.on('error', (data)=> {
+    console.log(data);
+});
+
+var req = https.request({port: 1336, host: 'https://example2.com', rejectUnauthorized: false}, function(){ 
```

Comment:
<!-- Comment managed by coverity-json-github-pr-commenter, do not modify!
4c7fb8aad3641a262a1dbe564a22d4d4
-->
# Coverity Issue - SSL certificate validation disabled
_Medium CWE-295_
The application disables certificate validation.

The "rejectUnauthorized" attribute is disabled.

Disabling certificate verification leads to a rogue certificate not being rejected when its signature cannot be verified by any of the certificate authorities, resulting in an insecure connection and a man-in-the-middle attack.Set "rejectUnauthorized" attribute to "true" to enable certificate validation.